.github/workflows/push.yml

```
@@ -18,3 +18,12 @@ jobs:
     with:
       build_type: branch
     secrets: inherit
+  trigger-pipeline:
+    runs-on: ubuntu-latest
+    needs: build-images
+    steps:
+      - name: Trigger Pipeline
+        env:
+          GH_TOKEN: ${{ secrets.WORKFLOW_TOKEN }}
+        run: |
+          gh workflow run build-and-publish-images.yml -R github.com/rapidsai/ci-imgs -r main --json '{"build_type":"branch"}'
```

```suggestion
          gh workflow run push.yml \
            --repo github.com/rapidsai/ci-imgs \
            --ref main
```

A few comments here:

- let's put the flags on multiple lines for readability
- let's use the long version of the flags for transparency (e.g. `--repo` instead of `-R`)
- according to the [docs](https://cli.github.com/manual/gh_workflow_run), `--json` is only used when passing arguments via `stdin`. `--field` seems like the correct flag here. though we actually don't need this flag at all due to my next point.
- I don't see any `workflow_dispatch` event triggers on the downstream workflow: https://github.com/rapidsai/ci-imgs/blob/main/.github/workflows/build-and-publish-images.yml
  - [this workflow](https://github.com/rapidsai/ci-imgs/blob/main/.github/workflows/push.yml) has `workflow_dispatch` and also doesn't require any fields like `branch` to be passed. that might be a better option to use instead